### DISCUSSION TITLE
How to trigger the copy content when change language for custom entity ?

### DISCUSSION THREAD

**Discussion by niji-rmejean** [sulu/sulu#6937] [Q&A] [answered]
Hi !

I'm trying to find the way I have to follow to trigger the "copy content" when I change language for my **translatable custom entities** like it's happen for content pages.

![image](https://user-images.githubusercontent.com/102292967/207388575-a786874e-b774-4aab-a081-6a2b0c502553.png)

I think there are some interfaces i can implement or something like that ...

Thanks a lot and have a good day !

**Comment by alexander-schranz** [comment, marked as answer]
In your endpoint you need to return an array of `availableLocales` and if it does not contain the current locale that overlay will be shown.

```json
{
    "id": "your-id",
    "availableLocales": ["en"]
}
```

https://github.com/sulu/sulu/blob/08863981d9e607dd1432db68eec2dac710732f74/src/Sulu/Bundle/AdminBundle/Resources/js/containers/Form/Form.js#L48-L50

**Reply by niji-rmejean** [reply]
Ok thanks for your answer ! I'll try that on next week end. Is there anything else I have to do for filling the fields ? Blocks content are also copied ?

**Reply by alexander-schranz** [reply]
@niji-rmejean The overlay will should if I remember correctlry trigger a post trigger action for copy-locale. The controller doing the copy for it you need to implement yourself as the UI doesnt know about how your data is stored and so how it need to be copied.

**Reply by niji-rmejean** [reply]
Worked great ! Thanks a lot !
